Those checks might be too fragile, and prevent to link static libs into shared
libs, while it's totally fine

--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -3460,7 +3460,7 @@ bsdi[[45]]*)
 
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
-  lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+  lt_cv_deplibs_check_method=pass_all
   lt_cv_file_magic_cmd='func_win32_libid'
   ;;
 
@@ -3469,11 +3469,11 @@ mingw* | pw32*)
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
   if ( file / ) >/dev/null 2>&1; then
-    lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+    lt_cv_deplibs_check_method=pass_all
     lt_cv_file_magic_cmd='func_win32_libid'
   else
     # Keep this pattern in sync with the one in func_win32_libid.
-    lt_cv_deplibs_check_method='file_magic file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)'
+    lt_cv_deplibs_check_method=pass_all
     lt_cv_file_magic_cmd='$OBJDUMP -f'
   fi
   ;;
